### ISSUE TITLE
Broken menu link on "Board of Veterans’ Appeals (BVA) hearing" page

### ISSUE THREAD

**Issue by stefaniefgray** [department-of-veterans-affairs/va.gov-cms#15864]
## Describe the defect

**Problem:** #content-broken-links has flagged the following page as containing a broken link: https://prod.cms.va.gov/disability/file-an-appeal/board-of-veterans-appeals
 
![image](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/9059428/7acf218c-6833-432c-91d7-c8a25bc225f7)

- The broken link in question is “Manage a legacy VA appeal” (an archived node), which is listed within the menu link breadcrumbs. See above screenshot -- the bad link is highlighted.

- If you click the “Manage a legacy VA appeal” link while viewing [this node](https://prod.cms.va.gov/disability/file-an-appeal/board-of-veterans-appeals) on prod, it goes to https://prod.cms.va.gov/disability/file-an-appeal, an un-published page (which is throwing the broken link alert) :x:

- But if you click into [this node](https://prod.cms.va.gov/disability/file-an-appeal/board-of-veterans-appeals) from the live site, the exact same breadcrumb menu link goes to https://www.va.gov/decision-reviews/legacy-appeals/ :white_check_mark: 

Note that the Board of Veterans Appeals page linked above is listed under the URL alias `/disability/file-an-appeal/` . 
On the live VA.gov, that path re-directs to: `https://www.va.gov/decision-reviews/legacy-appeals/`

## To Reproduce
1. Go to the ["Board of Veterans Appeals" page on prod](https://prod.cms.va.gov/disability/file-an-appeal/board-of-veterans-appeals) and click the “Manage a legacy VA appeal” link listed within the menu breadcrumbs. (See screenshot above)
2. Observe how this goes to an un-published page, which is creating the ongoing broken link alerts.
3. Go to the ["Board of Veterans Appeals" page on the live site](https://www.va.gov/disability/file-an-appeal/board-of-veterans-appeals) and click the “Manage a legacy VA appeal” link listed within the menu breadcrumbs. (See screenshot above)
4. You will be redirected to a published page.

 ## Additional context

Original message in #content-broken-links:
`{"path": "disability/file-an-appeal/board-of-veterans-appeals",`
`"linkErrors": [{"html": "<a href=\"/disability/file-an-appeal\" id=\"6bc608658fc266d4a1fdbe226d63d86b\">Manage a legacy VA appeal</a>",`
`"target": "[/disability/file-an-appeal](https://s3-us-gov-west-1.amazonaws.com/disability/file-an-appeal)"}]},`

## AC / Expected behavior

- [x] Changes are verified in Tugboat before modifying production
- [ ] CAIA is made aware and signs off on the change in Tugboat
- [ ] Menu link breadcrumbs on this page are accurate and no longer cause hourly broken link alerts

**Comment by stefaniefgray** [comment]
Slack thread: https://dsva.slack.com/archives/C52CL1PKQ/p1698089869966429

**Comment by chri5tia** [comment]
Not logged in (because the node is archived).
<img width="671" alt="Screenshot 2023-10-27 at 10 56 11 AM" src="https://github.com/department-of-veterans-affairs/va.gov-cms/assets/22764938/6e8b1400-cce2-4c56-80bf-5dbb08325189">

We can change the hierarchy of the child content to not be under the archived content.
Breadcrumb/hierarchy is determined by the URL alias defined:
<img width="387" alt="Screenshot 2023-10-27 at 11 14 54 AM" src="https://github.com/department-of-veterans-affairs/va.gov-cms/assets/22764938/eafe47ed-4556-4cec-ab26-b72365887d38">

`/disability/file-an-appeal/board-of-veterans-appeals`

Here: the fie-an-appeal has been archived: `[/disability/file-an-appeal`](https://prod.cms.va.gov/disability/file-an-appeal)
The fix is to get rid the file-an-appeal reference in any URL alias. This is strictly a content problem, not programmatic changes should be needed.

I would like to do or see this done:
1. Update the URL alias for https://prod.cms.va.gov/node/902/edit (https://prod.cms.va.gov/disability/file-an-appeal/board-of-veterans-appeals) to remove the 'file-an-appeal' from the url alias.
2. Update all of the URL aliases that mention this archived content

A programmatic way to create breadcrumbs automatically based on a hierarchy is to not manually define the URL alias but instead assign them to a menu structure and configure path auto to update the URL automatically. I am guess for the same reason that we have redirects being handled by the front end that we don't want to configure it this way.

**Comment by jilladams** [comment]
@chri5tia regarding the redirect that is in place here, can you find the code handling that redirect in redirects.yml? Per the Slack thread, it's a FE only redirect, so before we make changes, we should look at the ticket that requested it, and make sure we understand why it was done that way in the first place, if possible.

**Comment by chri5tia** [comment]
@jilladams I don't see how the fact that there is a redirect for the archived content has to do with what needs to be done in the CMS. The breadcrumbs will behave the same way regardless. We still need the redirect for people going to that page that doesn't exist anymore.

I will definitely take a look though.

**Comment by chri5tia** [comment]
@jilladams There is a redirect in place for 2 of the 3 links that use the archived node (file-an-appeal). The code in the redirect was added 3 years ago as a migration/rearrangement. It existed before that. 
https://github.com/department-of-veterans-affairs/devops/pull/7174

I feel like the fix that is needed in the CMS itself needs to be done regardless of redirects, which are in place because of published content and outdated references in the wild.

The fact that there is archived content in the CMS with children will continue to report broken breadcrumb links.The breadcrumbs will behave the same way regardless of the redirects. 

Here is what is in the redirects.yml code now:

```
  # https://github.com/department-of-veterans-affairs/va.gov-team/issues/8345
  - src: ^/disability/file-an-appeal/?$
    dest: /decision-reviews/legacy-appeals/
    matcher: "~*"

  - src: /disability/file-an-appeal/request-priority-review/
    dest: /decision-reviews/legacy-appeals/priority-review/
```

I feel like the content should be changed in the CMS to match the redirect requests. Maybe someone can educate me on that. But good housekeeping would be that these manually entered URL aliases are also updated in the CMS.

**Comment by jilladams** [comment]
Found the issue that generated the FE redirect request -- it's from the days of vets.gov, April 2020: https://github.com/department-of-veterans-affairs/va.gov-team/issues/8345. 

Again: I don't really understand what's happening in the backend, and because we've had so much confusion around breadcrumbs lately, my primary goal is just for us to know with 100% confidence that the CMS changes we want to make won't impact the existing redirects or breadcrumbs in the FE. Making the change in Tugboat and building a FE could help us prove that. Christia, if you already know that with confidence, could make the changes on Monday. (Would like to avoid Friday afternoon fire drills if anything goes wrong.)

**Comment by jilladams** [comment]
Listing of all published CLPs, with owners / secondary owners listed, based on reviewing the revision logs for each one.
[CLP Image aspect ratio Editors.xlsx](https://github.com/department-of-veterans-affairs/va.gov-cms/files/13218056/CLP.Image.aspect.ratio.Editors.xlsx)

**Comment by chri5tia** [comment]
## Engineering ride-along notes that you can ignore

There is a notice on the archived page that there is one node that links to it. 

> 1 link to unpublished page found
Linking to unpublished pages will result in broken links once this content is published. Prevent errors by omitting the following links, or updating their status to published.

> [Learn how to request the Advanced on Docket status] https://prod.cms.va.gov/disability/file-an-appeal/request-priority-review/
Node: 901

URL in question (from broken link report) is https://prod.cms.va.gov/disability/file-an-appeal/board-of-veterans-appeals
URL in question has a reminder to update the link:

> 1 redirected link found
Please update the following URLs to prevent the possibility of broken links in the future.

> Update [Find a Veterans Service Organization] https://www.va.gov/vso/ to its new location https://www.va.gov/ogc/recognizedvsos.asp
Node: 902

I need more education about the content architecture such as how URLs are defined. 

> Content Type: Benefits Detail Page
Owner: [VBA](https://prod.cms.va.gov/section/vba) » [VBA Benefits Hubs](https://prod.cms.va.gov/section/vba-benefits-hubs) » [Disability benefits hub](https://prod.cms.va.gov/section/vba/disability-benefits-hub)
VA.gov URL: https://www.va.gov/disability/file-an-appeal/board-of-veterans-appeals

This VA.gov URL references the broken node and matches the URL alias.
This is in block `block-entitymetadisplay` class `block block-va-gov-workflow-assignments block-va-gov-workflow-assignments-meta`. 

- [ ] see how this block works

Auto-alias is not enabled. URL Alias is manually defined as `/disability/file-an-appeal/board-of-veterans-appeals`.
Updating this to /disability/board-of-veterans-appeals in my local shows this in the entitymetadisplay block:
`VA.gov URL: https://va-gov-cms.ddev.site/static/disability/board-of-veterans-appeals (pending)`

I can surmise without looking at the block that it get's the URL from the URL alias field.

It was mentioned that these are not manually populated, but since auto-alias is off, I wonder how that is populated if not by the content creator.

This notice is also on the local and seems like it should be taken care of, I assume the redirect is in the CMS. Not sure if it's related but also seems outdated, as it is redirecting to an .asp page.

> 1 redirected link found
Please update the following URLs to prevent the possibility of broken links in the future.

Update [Find a Veterans Service Organization] https://www.va.gov/vso/ to its new location https://www.va.gov/ogc/recognizedvsos.asp
Node: 902

I don't want to grossly simplify this, but it still makes sense to me that the broken link is because the link is because the URL alias hasn't been updated to match the structure of content with the archived parent. What is the protocol for creating/using the URL alias field?

**Comment by chri5tia** [comment]
Manually edited the URL Alias in a demo environment to update the URL to not include the archived parent, which also updated the breadcrumb:
https://web-jmx2hyeh4iqszduwfruyivnqszlptcbq.demo.cms.va.gov/disability/board-of-veterans-appeals/

- [ ] Is there a test version of the broken link reports tied to the CMS? I'm sure that this quiets that since the broken breadcrumb is no longer present.
https://main-jmx2hyeh4iqszduwfruyivnqszlptcbq.demo.cms.va.gov/disability/board-of-veterans-appeals

Since this is what I understand to be the usual practice, I am not sure what to check against and need some insight/input.

@swirtSJW Any insight or advice?

- [ ] If not manually updating the URL aliases to match changed content architecture, than what _is_ the protocol/system in place?

**Comment by chri5tia** [comment]
Regarding my step to fix the alias in other nodes, they are archived, disregard.
> Update all of the URL aliases that mention this archived content. You can see the list here: https://prod.cms.va.gov/admin/config/search/path?search=file-an-appeal (right now there are only 2).

**Comment by jilladams** [comment]
For my brain: 

## child page
Node: https://prod.cms.va.gov/disability/file-an-appeal/board-of-veterans-appeals
Page: https://www.va.gov/disability/file-an-appeal/board-of-veterans-appeals/
Breadcrumb: [VA disability compensation](https://www.va.gov/disability) [Manage a legacy VA appeal](https://www.va.gov/disability/file-an-appeal) [Board of Veterans’ Appeals (BVA) hearing](https://www.va.gov/disability/file-an-appeal/board-of-veterans-appeals)

## live parent
Node: https://prod.cms.va.gov/disability/file-an-appeal
Page: https://www.va.gov/decision-reviews/legacy-appeals/
Breadcrumb: [Home](https://www.va.gov/)  [Decision reviews and appeals](https://www.va.gov/decision-reviews) [Manage a legacy VA appeal](https://www.va.gov/decision-reviews/legacy-appeals)

## previous parent
Node: https://prod.cms.va.gov/disability/file-an-appeal

## grandparent, not important but for reference
Node: https://prod.cms.va.gov/decision-reviews
Page: https://www.va.gov/decision-reviews/

**Comment by chri5tia** [comment]
1. Update the url alias of the child page URL alias to decision-reviews/legacy-appeals/board-of-veterans-appeals
2. Content release in tugboat
3. FE Breadcrumb should show "Manage a legacy VA appeal" that goes to the _new_ page without needing the redirect
4. Redirect for FE created for child page from `https://www.va.gov/disability/file-an-appeal/board-of-veterans-appeals/` to `https://www.va.gov/decision-reviews/legacy-appeals/board-of-veterans-appeals`

**Comment by jilladams** [comment]
Thread to CAIA for approval of the URL change / redirect proposal: https://dsva.slack.com/archives/C01K37HRUAH/p1699038995102449

**Comment by chri5tia** [comment]
In Demo environment:

1. Updated URL alias to match the parent's new URL
2. Checked that CMS breadcrumb goes to the new parent
3. Started a new content release

![Screenshot 2023-11-03 at 12 22 07 PM](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/22764938/b2567826-b78a-40a2-963a-e33f2a59f4ab)

![Screenshot 2023-11-03 at 12 22 15 PM](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/22764938/d534e728-6bb5-45ec-9e57-684b4bec2545)

![Screenshot 2023-11-03 at 12 22 25 PM](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/22764938/f5034506-3753-489f-8ae9-2e4bba205951)

![Screenshot 2023-11-03 at 12 22 41 PM](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/22764938/2a496dd7-0d0e-4e40-85a6-7633b5cd8653)

**Comment by jilladams** [comment]
CAIA would like Mikki to weigh in, and she's OOO til next Tues. So: this will be blocked til then, and likely crossover sprint boundary.

**Comment by chri5tia** [comment]
FE test build looks good: https://web-jmx2hyeh4iqszduwfruyivnqszlptcbq.demo.cms.va.gov/decision-reviews/legacy-appeals/board-of-veterans-appeals/

**Comment by jilladams** [comment]
https://dsva.slack.com/archives/C01K37HRUAH/p1699376477807459?thread_ts=1699038995.102449&cid=C01K37HRUAH
> Updating the URL and the redirect is what should be done. Hwever, I think this page and titling needs a content review because as it is, it is very duplicative to the core BVA page: https://www.va.gov/decision-reviews/board-appeal/veterans-law-judge-hearing/

https://dsva.slack.com/archives/C01K37HRUAH/p1699377383754849?thread_ts=1699038995.102449&cid=C01K37HRUAH
> [Content review should happen before URL change / redirect] to ensure that we even need to keep this page, or if its context changes enough where it impacts the URL, etc.

Now blocked on CAIA.

**Comment by jilladams** [comment]
Today CAIA approved making the interim URL change / redirect, to fix the broken link report: https://github.com/department-of-veterans-affairs/va.gov-team/issues/69398#issuecomment-1839347844
> Yes, go ahead and change the URL and implement the redirect. That will give the content team time to do the review and will eliminate noise for more important issues to be seen.

We may need to remove this redirect in future if the page gets archived after content review in https://github.com/department-of-veterans-affairs/va.gov-team/issues/69398. I went ahead and put up a PR ([here](https://github.com/department-of-veterans-affairs/devops/pull/13767)), but if it needs any changes or opens any questions, we should pull this into Sprint 99 and get engineer help to close it out. (FYI @FranECross )

**Comment by jilladams** [comment]
End of sprint: PR needs platform approval. Going to hold on making the CMS URL changes until I have a clear path to the redirect going out.

**Comment by jilladams** [comment]
Node path is updated & live after content release: https://www.va.gov/decision-reviews/legacy-appeals/board-of-veterans-appeals/

Revproxy deploy is requested: https://dsva.slack.com/archives/CBU0KDSB1/p1701820565853199

**Comment by jilladams** [comment]
In broken links channel, I see that the parent page here was referencing the old URL. Added a revision to https://prod.cms.va.gov/decision-reviews/legacy-appeals to use the new child page URL. 

If that doesn't get us sorted, @stefaniefgray could you point me to the broken links report for more info on what's still getting reported as broken? Thanks!

**Comment by jilladams** [comment]
That edit didn't take for some reason last week. Made it again today, looks correct in node revisions, so hopefully will be resolved after next content release.

**Comment by jilladams** [comment]
I'm no longer seeing these show up in the broken links report. Closing, but @stefaniefgray please feel free to reopen or cut a new ticket if you see more issues come through.
